subsys/bluetooth/mesh/net.c

```diff
@@ -653,15 +657,14 @@ static bool net_decrypt(struct bt_mesh_net_rx *rx, struct net_buf_simple *in,
 		return false;
 	}
 
-	if (rx->net_if == BT_MESH_NET_IF_ADV && msg_cache_match(out)) {
+	if (rx->net_if == BT_MESH_NET_IF_ADV && msg_cache_match(out, rx)) {
```

Comment:
I guess it is better to pass `net_idx` directly. `msg_cache_match` doesn't need to know about `bt_mesh_net_rx`.

Reply:
This is not addressed

Reply:
Haah ... changes were not staged :|